src/Microsoft.AspNet.WebUtilities/KeyValueAccumulator.cs

```diff
@@ -7,17 +7,16 @@
 
 namespace Microsoft.AspNet.WebUtilities
 {
-    public class KeyValueAccumulator
+    public struct KeyValueAccumulator
```

Comment:
Could do but would need the allocation of an outer dictionary, a closure; or some code rearranging? For example its used in the extension method [```QueryHelpers.ParseQuery```](https://github.com/aspnet/HttpAbstractions/blob/ceded805f32e3aca0066182fd96bbf171bd40574/src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs#L116)

You can avoid alloc with these changes by checking ```HasValues``` before calling ```GetResults()``` as is done in the [```FormFeature.InnerReadFormAsync```](https://github.com/aspnet/HttpAbstractions/pull/413/files#diff-25040e9a807f3ea89b0c6dc24fa8d50dR189)